cluster/rpc/server.go

```diff
@@ -179,3 +184,83 @@ func toRPCError(err error) error {
 	}
 	return status.Error(ec, err.Error())
 }
+
+// QuerierStream is experimental. QuerierStream is triggered when a querier node connects
+// to this metadata node via this bidirectional gRPC stream. We register the querier node
+// with the querier manager and send class tenant data updates to the querier node when
+// they appear on this querier's class tenant data updates channel.
+// We don't currently expect any messages from the querier node, but just use the existence
+// of the stream to keep the connection alive so we can send events to the querier node.
+// This function blocks until the stream is closed by the querier node.
+func (s *Server) QuerierStream(stream cmd.ClusterService_QuerierStreamServer) error {
+	// set up a querier and register it
+	q := querier.NewQuerier()
+	s.querierManager.Register(q)
+	defer s.querierManager.Unregister(q)
+
+	// streamClosed should receive a message when the stream is closed by the querier node,
+	// so we can close the goroutine that sends class tenant data updates.
+	streamClosed := make(chan struct{})
+	// returnErr should be set if there is an error in the goroutines below that should
+	// be returned to the caller.
+	var returnErr error
+
+	wg := sync.WaitGroup{}
+	wg.Add(2)
```

Comment:
I would add a comment here that add 2 because we will be waiting for 2 goroutines (one that receive messages and one that sends messages).

Also question from my end: is there a case where one goroutine will exit and not the to other ? For example the query node answer back with an error, we will exit the writing goroutine, but will we also clear the other one and boot the node ?

Reply:
I'll look more into this, I have handling for some basic goroutine exit scenarios but have not thought deeply about it